tools/js-sdk-release-tools/src/changelog/changelogGenerator.ts

```diff
@@ -9,98 +9,107 @@ import { TypeLiteralDeclaration } from "parse-ts-to-ast/build/declarations/TypeL
 import { TSExportedMetaData } from "./extractMetaData";
 import { SDKType } from "../common/types";
 import { logger } from "../utils/logger";
+import { detectBreakingChangesBetweenPackages, InlineDeclarationNameSetMessage, RuleMessage, RuleMessageKind } from "typescript-codegen-breaking-change-detector";
```

Comment:
I want to clarify my understanding for the scope of this pr: is it focused on RLC breaking detection or changelog detection?

Reply:
1. the decision is detecting breaking changes based on RLC inline usage. check it out: https://github.com/Azure/autorest.typescript/pull/1896#discussion_r1234682238
2. focused on RLC breaking detection, and the goal is to generate changelog
3. The cases you mention are covered by common breaking change detection in `changelogGenerator.ts`, it's not the goal of this pr. This pr is to post process the changelog, and remove the changes that should be considered as non-breaking in RLC
4. it's out of scope in this pr. it's a proposal, no decision for now

Reply:
1 & 2. sure then it seems no change compared with our previous decision.

3. Good to hear that, not sure we could add a path deletion in our [testing](https://github.com/albertxavier100/azure-sdk-for-js/pull/1/files), just corious what kind of breaking line would be, e.g Remove function `(path: "/widgets/{widgetName}", widgetName: string): GetWidget` in Routes?
4. got it.

Reply:
3. Looks like I should add a patch to detect routes. the current tool failed to detect any changes in routes in format: `   (path: "/deviceUpdate/{instanceId}/updates:import", instanceId: string): DeviceUpdateImportUpdate;`, due to it uses a terrible ast ask, it cannot pase routes.

Reply:
thanks for the confirmation, i also guess if the overloading could be detected or not. the overload could happen in isUnexpected and Routes with sharedRoute cases.

Reply:
Looks like the changes that failed to detect should be in base detector, since they are general typescript changes, not based on inline usages that makes some breaking changes to non breaking. added a todo here and plan to add it in base detector later.